admin/module/catalog-admin-module-sources/src/main/webapp/js/view/ModalSource.view.js

```diff
@@ -286,6 +287,9 @@ function (ich,Marionette,Backbone,ConfigurationEdit,Service,Utils,wreqr,_,$,moda
             var properties = config.get('properties');
             return properties.get('shortname') || properties.get('id');
         },
+        getPID: function(config) {
```

Comment:
I think that would be best because it just adds complexity. Or more idiomatically to backbone, you could add a `getPID` method directly to the model.